### PR TITLE
Fix UG bugs found during PED

### LINKED ISSUE #187: [PE-D][Tester E] Grammatical errors in 'Using this Guide' section
## Description
This sentence has some grammatical errors, with 2 verbs used. 

## Possible Solution 
Perhaps it can be amended to:
> This user guide is meant to ~~provide~~ guide users to use AssetBook and comprehensively document its features.

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: triciiaaa/ped#1

### LINKED ISSUE #189: [PE-D][Tester C] Inconsistent Formatting of Command
Some commands are given in a new line while others are given after a : , would it be better to standardise ?

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: MarcusGitty/ped#1

### LINKED ISSUE #190: [PE-D][Tester B] Inconsistent Prefix Parameter in UG
From reading through the UG, it is my belief that the prefix parameter for address should be a\ and asset should be A\. However, on Page 18 of the UG, the prefix might have been incorrectly entered.

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: Austintjh19/ped#2

### LINKED ISSUE #191: [PE-D][Tester A] Inconsistency in format and example in "Adding a Contact: add" in UG
Under "Adding a Contact: add" the format given is, where we can see that the prefix used is  "\\".

However, the example uses "/", which is a contradiction.

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: Apzure/ped#1

### LINKED ISSUE #197: [PE-D][Tester B] Inconsistent UG description for Phone Num Parameter
In the UG it mentions the Phone Number can take in any number of characters. However, this is inconsistent with the application, as it only allows for phone numbers with at least a character length of 3. This can be easily fix by updating the UG.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: Austintjh19/ped#4

### LINKED ISSUE #199: [PE-D][Tester E] User guide valid test case is invalid in actual JAR file
## Description
The valid test case provided in the user guide does not work on the actual JAR file. Looking at the various fields, it seems like the `o\` prefix should be `a\` since it refers to the address field. 

Additionally, the asset field seems to be using `a\` instead of `A\`. Seems unclear to me which is used for assets and address since both fields start with the letter A. 

## Possible Solution
To make it easier for users to remember, perhaps you can use a substring of the field for the prefix. For example:
`ad\311, Clementi Ave 2, #02-25` for address field
`as\screwdriver` for asset field

-------------
Labels: `type.DocumentationBug` `severity.Medium`
original: triciiaaa/ped#5

### LINKED ISSUE #200: [PE-D][Tester B] Phone Number parameter allows `+` to be entered anywhere.
I believe that it doesnt make sense to allow `+` sign to enter anywhere on the Phone Number. It is best to restrict it to the first character if utilised by the user. 

-------------
Labels: `type.FeatureFlaw` `severity.High`
original: Austintjh19/ped#5

### LINKED ISSUE #205: [PE-D][Tester E] Inconsistency in asset prefix
## Description
Looking at the overall User Guide, it seems like the prefix `\a` is used for the address field while `A\` is used for the asset field instead. However, this section on the User Guide shows that multiple assets can be added using the `a\` prefix instead.

-------------
Labels: `type.DocumentationBug` `severity.Medium`
original: triciiaaa/ped#7

### LINKED ISSUE #207: [PE-D][Tester C] Removing Assets via "Edit 1 a\" does not match the behavior in the UG
When i type "Edit 1 a\"

Expected: To remove all assets (based on UG)
Actual:  assets are not removed and encountered - "Addresses cannot contain '\', and should not be blank"

should change it to "Edit 1 A\" instead, with the A capitalised 

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: MarcusGitty/ped#4

### LINKED ISSUE #212: [PE-D][Tester B] Commands are not Case-sensitive
To the best of my knowledge, the UG does not mention that the command words are case-insensitive. However, in the application it is. As shown by ADD command being able to execute. This can be easily fixed by updating the UG. 

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: Austintjh19/ped#9

### LINKED ISSUE #213: Grammatical error in `Features` section of UG
Expected: `Parameters not specified by the command format entered by the user`

Actual:

image

### LINKED ISSUE #216: [PE-D][Tester C] Paste Keyboard on UG only matches the Windows shortcut, should include Mac Command + V
As MacOS is listed as a supported operating system, perhaps Command + V should be included as well. 

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: MarcusGitty/ped#6

### LINKED ISSUE #217: [PE-D][Tester D] Inconsistent documentation of find command
In the command summary it states `find KEYWORD [KEYWORD] ... `. Perhaps ensure consistancy by changing it to `find QUERY` as somewhere in the user guide uses `find QUERY`

Under command summary:

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: felixchanyy/ped#1

### LINKED ISSUE #227: [PE-D][Tester E] Weak validation on email domain
## Description
While the User Guide specifies that the email field should contain the format ‘local-part@domain', I am able to key in jd@gmail` which is possibly an incorrect domain. Perhaps it would be good to add stronger validation for valid domains used.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: triciiaaa/ped#11

### LINKED ISSUE #230: [PE-D][Tester A] UG Phrasing clarity issue with regards to assets in the edit command
In the "Editing a Contact: edit" command of the UG, it states:

"When editing assets, the existing assets of the person will be removed i.e adding of assets is not cumulative."

I believe the phrasing of this sentence can make it difficult to understand, and I personally do not understand what this sentence means.

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: Apzure/ped#10

### LINKED ISSUE #233: [PE-D][Tester A] Validity of Email Field
As shown in the image below, the UG states emails must be in the "valid email format". However, it is not clear what this format entails, and it does not seem to refer to being in "the format ‘local-part@domain’", since it is separated with an "and".

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: Apzure/ped#11

### LINKED ISSUE #236: [PE-D][Tester A] Database of Asset is unclear
In the description of the asset field, it is stated that "Assets must have unique names. If the asset already exists in the database, the existing
asset will be linked instead of a new asset."

It may not be clear as to what the database refers to. Is it the database within the app itself, or is it linked to some online database? 

Among non-technical users, which the UG should be targeting, it is possible that they are unfamiliar with the "database" term, and this could confuse them.

In addition, this was the only instance the term "database" was used in the UG, with no clarification.

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: Apzure/ped#12

### LINKED ISSUE #237: [PE-D][Tester A] Existing asset being linked vs a new asset being linked
In the description of the asset field, it is stated "Assets must have unique names. If the asset already exists in the database, the existing asset will be linked instead of a new asset.".

The implications of an existing asset being linked vs a new asset being linked is unclear to the user. It is also unclear as to what it means to be "linked", as that is a term with particular meaning for your app that is not specified.

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: Apzure/ped#13

### LINKED ISSUE #239: [PE-D][Tester A] Clarify the term "Modifying"
In the "Undo Command", it states that it Undoes the last modifying command.

However, it is not clear as to what makes a command "modifying".

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: Apzure/ped#15

### LINKED ISSUE #240: [PE-D][Tester A] Lack of examples for undo command
There are no examples provided for the undo command.

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: Apzure/ped#16

### LINKED ISSUE #241: [PE-D][Tester A] Lack of Examples for Redo Command
No examples provide for the redo command.

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: Apzure/ped#17

### LINKED ISSUE #246: [PE-D][Tester D] Missing Help Command in UG
Perhaps include help command documentation in user guide so that user will know all the available commands.

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: felixchanyy/ped#12

### LINKED ISSUE #251: [PE-D][Tester D] Copy command PREFIX not shown to user in UG
Perhaps can show available prefix together with copy command in user guide so that user can know what prefix are there.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: felixchanyy/ped#15

### LINKED ISSUE #253: Missing hyperlink in table of content
Missing Editing the data file hyperlink in table of content. Perhaps can include a hyperlink in the toc to help user easy navigate to there.

image

image

https://github.com/felixchanyy/ped/issues/16

### LINKED ISSUE #256: Update app version for v1.4 release

### PR DESCRIPTION
**Pull request by yisiox**
+ Fixes #253
+ Fixes #251 
+ Fixes #246
+ Fixes #237  
+ Fixes #236
+ Fixes #216
+ Fixes #213 
+ Fixes #212
+ Fixes #207
+ Fixes #205
+ Fixes #191
+ Fixes #187
+ Fixes #253
+ Fixes #199
+ Fixes #217
+ Fixes #189
+ Fixes #190
+ Fixes #197
+ Fixes #239
+ Fixes #240
+ Fixes #241
+ Fixes #230 
+ Fixes #200 
+ Fixes #233 
+ Fixes #227

### CODE REVIEW

**Comment by aureliony** [comment]
TODO: Write user-friendly explanation for SUBSTRING

**Comment by aureliony** [comment]
Placing saving or editing the data file under the "miscellaneous shortcuts" doesn't make sense.

File: docs/UserGuide.md [hunk 1]
```diff
@@ -80,22 +80,24 @@ or skip to the [command summary](#command-summary).
   + [Navigating the GUI](#navigating-the-gui)
   + [Tutorials](#tutorials)
 + [Features](#features)
-  + [Modifying Contact Data](#adding-a-contact-add)
+  + [Modifying Contact Data](#modifying-contact-data)
```
Comment (aureliony) [inline docs/UserGuide.md:83]: Perhaps it would be better to name this section "Modifying Commands"?

**Review by aureliony — CHANGES_REQUESTED** [review]

**Comment by bryanyee33** [comment]
> TODO: Write user-friendly explanation for SUBSTRING

Solved by rephrasing explanation instead.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -315,10 +316,16 @@ Colored text that look like the following describe the format of a command.
...
 A valid input by the user corresponding to the above will be
 
-`add n\John Doe e\johndoe@example.com o\574 Ang Mo Kio Ave 10 p\12345678 a\L293D`
+`add n\John Doe e\johndoe@example.com a\574 Ang Mo Kio Ave 10 p\12345678 A\L293D`
 
 <div style="page-break-after: always;"></div>
 
+<box type="tip" seamless>
+
+Command words are not case-sensitive, i.e. add = Add = ADD.
```
Comment (aureliony) [inline docs/UserGuide.md:325]: Should we write this in English instead of using mathematical symbols?

**Review by aureliony — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md (hunk 2 above)
Comment (bryanyee33) [reply docs/UserGuide.md]: Yes, it would be more readable.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -410,7 +421,8 @@ If the example was not executed successfully, the proper syntax of the `add` com
 `PHONE`
 * Phone number of the contact.
 * Only digits, '+', '-', ',' and spaces are allowed.
-* Any number of these characters are allowed.
+* The phone number must contain at least 3 characters.
+* Example: `+65-8765-4321, +97 6223 5555`.
```
Comment (aureliony) [inline docs/UserGuide.md:425]: Currently, phone numbers are not validated extensively, e.g. `+++` is a valid input. Should we mention that as well?

**Review by aureliony — CHANGES_REQUESTED** [review]

**Comment by aureliony** [comment]
TODO: replace remaining traces of "person" with "contact"

File: docs/UserGuide.md (hunk 1 above)
Comment (bryanyee33) [reply docs/UserGuide.md]: Yes, it has been changed.

File: docs/UserGuide.md (hunk 3 above)
Comment (bryanyee33) [reply docs/UserGuide.md:425]: No, as it is not a bug to allow seemingly unsuitable values.

image

**Review by aureliony — APPROVED** [review]
LGTM

**Review by bryanyee33 — APPROVED** [review]
LGTM

**Review by darkvoid32 — APPROVED** [review]
LGTM

**Pull request merged**